samples/drivers/video/pipeline/src/main.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/video.h>
+#include <zephyr/drivers/video-controls.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util_macro.h>
+
+#include <mp.h>
+
+LOG_MODULE_REGISTER(main, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define PIPELINE_ID    0
+#define CAM_SRC_ID     1
+#define CAPS_FILTER_ID 2
+#define VID_TRANS_ID   3
+#define DISP_SINK_ID   4
+
+int main(void)
+{
+	int ret;
+	struct mp_element *pipeline, *source, *sink;
+	__maybe_unused struct mp_element *caps_filter = NULL, *transform = NULL;
+
+	/* Create a new pipeline */
+	pipeline = mp_pipeline_new(PIPELINE_ID);
+	if (pipeline == NULL) {
+		goto err;
+	}
+
+	/* Camera source element */
+	source = mp_element_factory_create(MP_ZVID_SRC_ELEM, CAM_SRC_ID);
+	if (source == NULL) {
+		goto err;
+	}
+
+	struct video_rect __maybe_unused crop = {
+		CONFIG_VIDEO_SOURCE_CROP_LEFT, CONFIG_VIDEO_SOURCE_CROP_TOP,
+		CONFIG_VIDEO_SOURCE_CROP_WIDTH, CONFIG_VIDEO_SOURCE_CROP_HEIGHT};
+
+	ret = mp_object_set_properties(
+		MP_OBJECT(source),
+		COND_CODE_0(CONFIG_PROP_NUM_BUFS, (), (PROP_NUM_BUFS, CONFIG_PROP_NUM_BUFS,))
+				 COND_CODE_0(CONFIG_VIDEO_SOURCE_CROP_WIDTH, (), (PROP_ZVID_CROP, &crop,))
```

Comment:
it's possible to wrap some code in `/* clang-format off */` and `/* clang-format on */` in case it gives something too odd, to keep some manual formatting.